### ISSUE TITLE
Print artifact digest after build process is completed

### ISSUE THREAD

**Issue by elad-pticha** [pypa/build#789]
I think it would be beneficial to add artifact digests to the Successfully built message. For instance:
``` bash
Successfully built test-0.0.0.tar.gz@sha256:960b9adda66023aed657c0da9626a6b8de71e433843181a5397431465adb57a7 and test-0.0.0-py3-none-any.whl@sha256:c856716babc6d603769f6b4f1a7122a61b870e7f627b0e2b4aa8f48b712c7770
```

![image](https://github.com/pypa/build/assets/124190990/1cb1f36c-e13e-4045-b9a3-c85218fed4eb)

Including the digest information strengthens the link between the build system and the package repository.

Let's take, for example, a GitHub Actions pipeline that builds and pushes packages to PyPI. Users can check the package's hash in the package repository but have no idea if that artifact is really built from the source repository it claims to be related to. If a digest is printed at the end of the build process, it could indicate that the package in the package registry has been created in a specific build pipeline.

This also relates to the world of [SLSA](https://slsa.dev/) and artifact integrity, allowing users to link between an artifact to its source.

**Comment by elad-pticha** [comment]
If you think this is a good idea, I can submit a PR to implement it.

**Comment by henryiii** [comment]
I’d like to have a machine readable output (json probably), but it can’t be printed, since it is problematic to try to pick it out of stdout. Probably an argument that allows writing the built package info to a stream or file. Having the name(s) and maybe other info is useful too when building pipelines.

**Comment by henryiii** [comment]
If you had such a machine readable output, then you could process it and produce the digest in your preferred format. No need to make a list of formats.

**Comment by elad-pticha** [comment]
I agree that build logs are not a place to put security information and that this does not come any close to creating a proper SLSA attestation, but maybe it is better than having no information at all for projects that don't produce attestations.

Another option is that this issue should be under the [twine](https://github.com/pypa/twine) repository and that the digest should be printed after the package has been pushed.
